lib/plugins/cors.js

```diff
@@ -35,16 +35,25 @@ var AC_EXPOSE_HEADERS = 'Access-Control-Expose-Headers';
 
 ///--- Internal Functions
 
+// turn a string defining an allowed domain into a regex
+function toRegExp(str) {
+    if (str instanceof RegExp) {
+        return str;
+    }
+    var regexStr = str
+      // escape all the special characters, except *
+      .replace(/[\-\[\]\/\{\}\(\)\+\?\.\\\^\$\|]/g, '\\$&')
+      // replace all instances of * with a proper regex wildcard
+      .replace(/\*/g, '.*');
+
+    return new RegExp('^(https?://)?' + regexStr + '$', 'i');
+}
+
 function matchOrigin(req, origins) {
     var origin = req.headers.origin;
 
     function belongs(o) {
-        if (origin === o || o === '*') {
-            origin = o;
-            return (true);
-        }
-
-        return (false);
+        return o.test(origin);
```

Comment:
I was following that approach in the beginning. The problem is that for values like `*.example.com` , or also simply `www.example.com` (any protocol), I need to convert the string to a regex. So at least some values in the array need to be turned into a regular expression. If I do an `if (o instanceof RegExp) { ... }` (and I would need to do it more than once, considering the preprocessing), then the code starts to look messy, so I decided to turn every value into a regex instead. If you prefer, I can go back to my initial implementation and differentiate between previous implementation (when in format `http://fixed.url` or `*`) and new one (`*.any.subdomain`, `any.protocol`, or `/regex/`).